src/pkg/bundle/deploy.go

```diff
@@ -541,3 +485,200 @@ func formFilePath(anchorPath string, filePath string) (string, error) {
+		return false
+	}
+	return true
+}
+
+type PkgView struct {
+	meta      map[string]string
+	overrides map[string]interface{}
+}
+
+// formPkgViews creates a unique pre deploy view of each packages set overrides and Zarf variables
+func formPkgViews(b *Bundle) []PkgView {
+	var pkgViews []PkgView
+	for _, pkg := range b.bundle.Packages {
+		pkgMeta := map[string]string{"name": pkg.Name, "ref": pkg.Ref}
+		variables := make([]interface{}, 0)
+
+		if pkg.Repository != "" {
+			pkgMeta["repo"] = pkg.Repository
+		} else {
+			pkgMeta["path"] = pkg.Path
+		}
+
+		// process overrides and variables to get values
+		_, pkgOvers := b.loadVariables(pkg, nil)
+		valuesOverrides, _, _ := b.loadChartOverrides(pkg, map[string]string{})
+
+		// updated on each chart iteration
+		filteredVars := pkgOvers
+
+		for compName, component := range pkg.Overrides {
+			for chartName, chart := range component {
+				// filter out the chart.Variables so left with Zarf Variables
+				filteredVars = filterOverrides(chart.Variables, filteredVars)
+
+				processedVars := valuesOverrides[compName][chartName]
+				if processedVars == nil {
+					continue
+				}
+
+				chartVars := map[string]interface{}{}
+
+				for _, v := range chart.Variables {
+
+					// Mask potentially sensitive variables
+					if v.Type == "file" || v.Source == valuesources.Env {
+						chartVars[v.Name] = "****"
+						continue
+					}
+
+					// handle complex paths: var.helm.path = { var: { helm: { path: val } } }
+					if strings.Contains(v.Path, ".") {
+						paths := strings.Split(v.Path, ".")
+
+						// set initial entry so iterations through paths can hold next key value pair until final value is found,
+						// removing the entry if map[path] returns nil
+						chartVars[v.Name] = processedVars
+						for _, path := range paths {
+
+							val, notNil := chartVars[v.Name].(map[string]interface{})[path]
+							if !notNil {
+								// delete previously set entry of v.Name and exit loop
+								delete(chartVars, v.Name)
+								break
+							}
+
+							chartVars[v.Name] = val
+						}
+					} else {
+						if processedVars[v.Path] == nil {
+							continue
+						}
+
+						chartVars[v.Name] = processedVars[v.Path]
+					}
+				}
+
+				if len(chartVars) > 0 {
+					variables = append(variables, map[string]map[string]interface{}{chartName: {"variables": chartVars}})
+				}
+			}
+		}
+
+		for key, fv := range filteredVars {
+			// config is location of uds-config.yaml, not a variable
+			if key != "CONFIG" {
+				// Mask potentially secret ENV vars
+				if fv.source == valuesources.Env {
+					fv.value = "****"
+				}
+				variables = append(variables, map[string]string{key: fv.value})
+			}
+		}
+
+		pkgViews = append(pkgViews, PkgView{meta: pkgMeta, overrides: map[string]interface{}{"overrides": variables}})
+	}
+	return pkgViews
+}
+
+func filterOverrides(chartVars []types.BundleChartVariable, varMap map[string]overrideView) map[string]overrideView {
+	varMapCopy := varMap
```

Comment:
Maps are passed by reference in Go. This means that `varMapCopy` points to the same underlying data as `varMap`, and changes to `varMapCopy` affect `varMap`. 

Also, what exactly are we filtering here? Looks like sometimes values are an empty string?

Reply:
hmm. ok yeah so we'd have to make a deep copy. even though i prefer immutability i don't know if it matters much here and is worth the effort / performance. maybe the fact that we're returning and resetting the `filteredVars` is enough to indicate the map is being updated. 

so we're filtering out entries generated during `loadVariables` that are associated with a bundle override variable. We should be left with only the Zarf Variables that were set. if we find an actual entry (not empty string)based on the bundle override variable name then we want to delete it. i'll work on making this more explicit.